### ISSUE TITLE
Doesn't handle `@!macro` definition and usage in different files

### ISSUE THREAD

**Issue by zaben903** [mensfeld/yard-lint#64]
When using an `@!macro` in another file to where the macro was originally defined `UndocumentedMethodArgument` is still raised.

The following will raise `UndocumentedMethodArgument:` in class `B`. Class `A` works as expected.

```ruby 
# a.rb

class A
  # @!macro testing
  #   @param some_key [String] Some Key

  # @macro testing
  def self.a_method(some_key)
    puts some_key
  end
end
```

```ruby
# b.rb

require_relative "a"

class B
  # @macro testing
  def self.b_method(some_key)
    A.a_method
  end
end
```
